peer_review_data/main.py

```diff
@@ -99,103 +96,116 @@ def saveAssessmentsAndComments(
-
-    if len(canvasAssignmentRubric.assessments) == 0:
-        LOGGER.info(
-            f'Skipping assignment ({ASSIGNMENT_ID}) in course ({COURSE_ID}): '
-            'No peer reviews ("assessments") were found.')
-        sys.exit()
-
-    LOGGER.info(f'Assignment ({ASSIGNMENT_ID}) in course ({COURSE_ID}) has '
-                'peer reviews ("assessments")…')
-
-    if not courseSaved:
-        courseSaved, course = saveCourseAndUsers(canvasCourse)
-
-    assignment: models.Assignment = models.Assignment.fromCanvasAssignment(
-        canvasAssignment)
-    LOGGER.info(f'Saving {assignment}…')
-    assignment.save()
-
-    LOGGER.info(f'Saving submissions for {assignment}…')
-    saveSubmissions(canvasAssignment)
-
-    LOGGER.info('Saving rubric and criteria…')
-    rubric, criteria = saveRubricAndCriteria(canvasAssignmentRubric,
-                                             canvasAssignment)
-
-    LOGGER.info('Saving assessments and comments…')
-    saveAssessmentsAndComments(canvasAssignmentRubric.assessments)
+    courseId: str
+    for courseId in [c.strip() for c in config.COURSE_IDS_CSV.split(',')]:
+        canvasCourse: CanvasCourse = canvas.get_course(courseId)
+        LOGGER.info(f'Found course ({canvasCourse.id}): "{canvasCourse.name}"')
+        courseSaved = False
+        course: models.Course | None = None
+
+        canvasAssignments: List[CanvasAssignment] = \
+            canvasCourse.get_assignments()
+
+        canvasAssignment: CanvasAssignment
+        # Iterate over assignments which have peer reviews
+        for canvasAssignment in filter(
+                lambda a: a.peer_reviews, canvasAssignments):
+            LOGGER.info(
+                f'Found peer reviewed assignment ({canvasAssignment.id}): '
+                f'"{canvasAssignment.name}"')
+
+            assignmentRubricId: int = canvasAssignment.rubric_settings.get(
+                'id')
+            LOGGER.debug(f'Assignment ({canvasAssignment.id}) has '
+                         f'rubric ID ({assignmentRubricId})')
+
+            canvasAssignmentRubric: CanvasRubric = canvasCourse.get_rubric(
+                assignmentRubricId, include='assessments', style='full')
+
+            if not hasattr(canvasAssignmentRubric, 'assessments'):
+                LOGGER.info(f'Skipping assignment ({canvasAssignment.id}) in '
+                            f'course ({courseId}): '
+                            'Not configured for peer reviews ("assessments").')
+                continue
+
+            LOGGER.info(f'Assignment ({canvasAssignment.id}) '
+                        f'in course ({courseId}) is '
+                        'configured for peer reviews ("assessments")…')
+
+            if len(canvasAssignmentRubric.assessments) == 0:
+                LOGGER.info(
+                    f'Skipping assignment ({canvasAssignment.id}) '
+                    f'in course ({courseId}): '
+                    'No peer reviews ("assessments") were found.')
```

Comment:
Should the error message be more specific to include the word 'rubric' inside, as 'No peer reviews rubrics assessments were found'?

Reply:
No.  That would be a confusing error message.  The condition isn't looking for rubrics, it's looking for peer reviews.  Or, as Canvas refers to them "assessments".